assets/app/scripts/directives/popups.js

```diff
@@ -86,15 +86,21 @@ angular.module('openshiftConsole')
       restrict: 'E',
       scope: {
         route: '=',
-        service: '='
+        service: '=',
+        warnings: '=' // instead of route and service, can provide an existing set of warnings
       },
       link: function($scope) {
         var updateWarnings = function() {
           var warnings = RoutesService.getRouteWarnings($scope.route, $scope.service);
           $scope.content = warnings.join('<br>');
         };
-        $scope.$watch('route', updateWarnings, true);
-        $scope.$watch('service', updateWarnings, true);
+        if ($scope.route) {
+          $scope.$watch('route', updateWarnings, true);
+          $scope.$watch('service', updateWarnings, true);
+        }
+        else if ($scope.warnings) {
+          $scope.content = $scope.warnings.join('<br>');
```

Comment:
Yeah that is probably it.  I could not generate the warnings if the service
hasn't loaded yet.  Since the route shouldn't even show up if the service
isn't there.

On Wed, Feb 17, 2016 at 2:56 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/directives/popups.js
> <https://github.com/openshift/origin/pull/7325#discussion_r53221612>:
>
> >        },
> >        link: function($scope) {
> >          var updateWarnings = function() {
> >            var warnings = RoutesService.getRouteWarnings($scope.route, $scope.service);
> >            $scope.content = warnings.join('<br>');
> >          };
> > -        $scope.$watch('route', updateWarnings, true);
> > -        $scope.$watch('service', updateWarnings, true);
> > +        if ($scope.route) {
> > +          $scope.$watch('route', updateWarnings, true);
> > +          $scope.$watch('service', updateWarnings, true);
> > +        }
> > +        else if ($scope.warnings) {
> > +          $scope.content = $scope.warnings.join('<br>');
>
> Is this the bug? I suspect you need a $watch here. Otherwise, the routes
> can load before the services, causing the overview warning. Would be nice
> if we can avoid the warning icon from flickering when that happens.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7325/files#r53221612>.
>